### PR TITLE
FAT-32 Implement API Karate tests for MODCAL

### PR DESCRIPTION
**Pull request by roman-barannyk**
Resolves: [FAT-32](https://issues.folio.org/browse/FAT-32)

### CODE REVIEW

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 1]
```diff
@@ -4,112 +4,384 @@ Feature: Calendar periods
     * url baseUrl
     * callonce login testUser
     * configure headers = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': 'application/json, text/plain' }
+    * def startDate = '2030-08-01';
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:7]: Push to 2120

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 2]
```diff
@@ -4,112 +4,384 @@ Feature: Calendar periods
...
+    * def periodName = 'Test period'
+    * def createPeriodRequest = read('samples/createPeriod.json')
+
+    Given path 'calendar/periods/' + servicePointId + '/period'
+    And request createPeriodRequest
+    When method POST
+    Then status 201
+
+    Given path 'calendar', 'periods'
+    And param endDate = '2030-08-03'
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:64]: Year

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 3]
```diff
@@ -4,112 +4,384 @@ Feature: Calendar periods
...
+    * def endExceptionDate = '2030-08-06'
+    * def createExceptionRequest = read('samples/createException.json')
+    And request createExceptionRequest
+    When method POST
+    Then status 201
+
+    Given path 'calendar', 'periods'
+    When method GET
+    Then status 200
+    And match $.openingPeriods[4].date == '2030-08-06T00:00:00.000+00:00'
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:123]: Year

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 4]
```diff
@@ -4,112 +4,384 @@ Feature: Calendar periods
...
+    And match $.openingPeriods[4].openingDay.exceptional == true
 
-  @Undefined
   Scenario: GET library hours period for service point with 200 on success
-    * print 'undefined'
+    * def servicePointId = call uuid1
+
+    Given path 'calendar/periods/' + servicePointId + '/period'
+    When method GET
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:130]: Validate the response

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 5]
```diff
@@ -4,112 +4,384 @@ Feature: Calendar periods
...
+    Then status 200
 
-  @Undefined
   Scenario: GET library hours period for service point with opening days and 200 on success
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:133]: Add withOpeningDays query param

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 6]
```diff
@@ -4,112 +4,384 @@ Feature: Calendar periods
...
-    * print 'undefined'
+    * def periodId = call uuid1
+    * def servicePointId = call uuid1
+    * def periodName = 'Test period'
+    * def createPeriodRequest = read('samples/createPeriod.json')
 
-  @Undefined
-  Scenario: GET library hours period for service point including past openings and with 200 on success
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:42]: Revert this, add showPast query parameter

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 7]
```diff
@@ -4,112 +4,384 @@ Feature: Calendar periods
...
-    * print 'undefined'
+    Given path 'calendar/periods/' + servicePointId + '/period'
+    And request createPeriodRequest
+    When method POST
+    Then status 201
 
-  @Undefined
-  Scenario: GET library hours period for service point with exceptional hours and 200 on success
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:46]: Revert, test showExceptional parameter

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature (hunk 1 above)
Comment (roman-barannyk) [reply mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature]: done

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature (hunk 2 above)
Comment (roman-barannyk) [reply mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature]: done

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature (hunk 4 above)
Comment (roman-barannyk) [reply mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature]: done

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature (hunk 5 above)
Comment (roman-barannyk) [reply mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:133]: done

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature (hunk 6 above)
Comment (roman-barannyk) [reply mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:42]: done

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature (hunk 7 above)
Comment (roman-barannyk) [reply mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:46]: done

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 8]
```diff
@@ -4,112 +4,436 @@ Feature: Calendar periods
...
+    * def createPeriodRequest = read('samples/createPeriod.json')
+
+    Given path 'calendar/periods/' + servicePointIdUrl + '/period'
+    And request createPeriodRequest
+    When method POST
+    Then status 400
+    And match $ contains 'Not valid json object. Missing field(s)'
 
-  @Undefined
   Scenario: POST calendar period by service point id should return 422 if periods overlap
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:300]: Use different start and end date for two periods

File: mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature [hunk 9]
```diff
@@ -4,112 +4,436 @@ Feature: Calendar periods
...
+
+    Given path 'calendar/periods/' + servicePointId + '/period/' + periodId
+    And request updatedInvalidPeriodRequest
+    When method PUT
+    Then status 422
+    And match $.errors[0].message == 'must not be null'
+    And match $.errors[0].parameters[0].key == 'startDate'
 
-  @Undefined
   Scenario: PUT library hours period by id should return 422 if periods overlap
```
Comment (alexanderkurash) [inline mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature:387]: Same here

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**